src/Workspaces/Core/Portable/Rename/IRenameRewriterLanguageService.cs

```diff
@@ -149,5 +157,143 @@ internal abstract class AbstractRenameRewriterLanguageService : IRenameRewriterL
+        /// class Hello
+        /// {
+        ///     void World()
+        ///     {
+        ///         var hello = "Hello World".
+        ///     }
+        /// }
+        /// If rename class 'Hello' to 'Hello2', void 'World' to 'World2',
+        /// <param name="locationRenameContexts"/> would contain the subspan info within string "Hello World".
+        /// The output dictionary woud looks like
+        /// {
+        ///     "0-4" : ("Hello", "Hello2"),
+        ///     "6-10" : ("World", "World2")
+        /// }
+        /// </summary>
+        internal static ImmutableSortedDictionary<TextSpan, (string replacementText, string matchText)> CreateSubSpanToReplacementTextInfoDictionary(
+            HashSet<LocationRenameContext> locationRenameContexts)
+        {
+            var subSpanToReplacementTextBuilder = ImmutableSortedDictionary.CreateBuilder<TextSpan, (string replacementText, string matchText)>();
+            foreach (var context in locationRenameContexts)
+            {
+                var renameLocation = context.RenameLocation;
+                var location = renameLocation.Location;
+                var replacementText = context.ReplacementText;
+                var originalText = context.OriginalText;
+                if (location.IsInSource && renameLocation.IsRenameInStringOrComment)
+                {
+                    var sourceSpan = location.SourceSpan;
+
+                    // SourceSpan should be a part of the containing location.
+                    RoslynDebug.Assert(sourceSpan.Start >= renameLocation.ContainingLocationForStringOrComment.Start);
+                    RoslynDebug.Assert(sourceSpan.End <= renameLocation.ContainingLocationForStringOrComment.End);
+
+                    // Calculate the relative postion within the containg location.
+                    var subSpan = new TextSpan(
+                        sourceSpan.Start - renameLocation.ContainingLocationForStringOrComment.Start,
+                        sourceSpan.Length);
+
+                    if (!subSpanToReplacementTextBuilder.TryGetValue(subSpan, out var replacementTextAndOriginalText))
+                    {
+                        subSpanToReplacementTextBuilder[subSpan] = (replacementText, originalText);
+                    }
+                    else if (!replacementTextAndOriginalText.Equals((replacementText, originalText)))
+                    {
+                        // Two symbol tries to rename a same subspan,
+                        // Exmaple:
+                        //      // Comment Hello
+                        // class Hello
+                        // {
+                        //    
+                        // }
+                        // class World
+                        // {
+                        //    void Hello() { }
+                        // }
+                        // If try to rename both 'class Hello' to 'Bar' and 'void Hello()' to 'Goo'. So both of them will try to rename
```

Comment:
So this should not be hit for the single symbol scenario. (Because there would be only one replacement text)
For multiple symbols, I would like to have a sanity check to first prevent this once we get the RenameLocations from the FAR engine [here](http://index/?leftProject=Microsoft.CodeAnalysis.Workspaces&leftSymbol=rbru0yuiedus)
(Checking the text span to see if it would be renamed to different replacement text)

And use the `ErrorMessage` [here](http://index/?query=ConflictResolut&rightProject=Microsoft.CodeAnalysis.Workspaces&file=Rename%5CConflictResolution.cs&line=15) to let user know we hit an error when try to a text span to different text

Reply:
I guess in the case though where there is a conflict, if the user wants to proceed though do we still need to allow that?

Reply:
Maybe we can introduce another conflict kind here? https://sourceroslyn.io/#Microsoft.CodeAnalysis.Workspaces/Rename/ConflictEngine/RelatedLocationType.cs,10
Suppose it is called 'TextOverlappingConflict'. For each of these places, we might just pick a random replacement text and annotate it?

Reply:
I feel this would be the work of conflict resolution. 
From the RenamerRewriter perspective, it just wants to get some instructions about how to rewrite the syntax tree. I feel we should still throw a renamer-specific exception here to indicate the input is wrong.